modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,241 @@
+import {Meta, ArgsTable} from '@storybook/addon-docs';
+import {Box} from '@workday/canvas-kit-react/layout';
+import {useResizeObserver, useResponsiveContainerStyles} from '@workday/canvas-kit-react/common';
+import {ResponsiveContainer} from './examples/ResponsiveStyles';
+import {ResponsiveViewport} from './examples/ResponsiveViewport';
+
+<Meta title="Hooks and Utilities/Responsive Styling" />
+
+# Responsive Styling
+
+Responsive styling is an approach to web development that adjusts layout and appearance based on the
+device's viewport size or an element's container size.
+
+## What is Responsive Styling?
+
+Responsive Styling it is a term used to describe an approach to web design or a set of best
+practices, used to create a layout that can _respond_ to the device being used to view the content.
+
+Read more about the history of
+[Responsive Design](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Responsive_Design).
+
+### Container vs Viewport
+
+There are two ways we will look at responsive styling below. They are container-based and
+viewport-based responsive styling.
+
+> Using container-based responsive styling, we will measure the size of the container the element is
+> in and apply styles based on the size of that container.
+
+> Using viewport-based responsive styling, we will measure the size of the user's viewport and apply
+> styles based on the size of that viewport.
+
+If you are measuring a user's viewport size, then you should be using viewport-based responsive
+styling.
+
+If you only want to measure the size of a container, then you should be using container-based
+responsive styling.
+
+> **Note**: A container's size can be affected by surrounding containers or the viewport size.
```

Comment:
I updated this content to remove some of the repetitive language and to segue into the next topic of breakpoints.
```suggestion
### Container-based vs. Viewport-based Responsive Styling

There are two approaches to responsive styling:

1. With **container-based** styling, we measure the size of the container an element is in and apply
   styles based on the size of that container. Note that a container's size can be affected by
   surrounding containers or the viewport size.
2. With **viewport-based** styling, we measure the size of the user's viewport and apply styles
   based on the size of that viewport.

Both approaches use breakpoints.
```